### DISCUSSION TITLE
New patterns: What to write, what to link, and what to skip?

### DISCUSSION THREAD

**Discussion by spier** [InnerSourceCommons/InnerSourcePatterns#566] [General]
When discussing ideas for potential new InnerSource Patterns[^1], we sometimes wonder which material to write ourselves, what external 3rd party material to link to, and what to skip (i.e. patterns we don't want to write at all). 

Now how do we make this judgement call? I am not aware of any documentation around this. Therefore I am starting this conversation to collect input about it. If the conversations leads to conclusive answers that are easy enough to follow, we may convert this into written documentation elsewhere.

Btw I assume that this problem is generally applicable throughout the InnerSource Commons, and not strictly limited to our Patterns.

This is certainly a discussion on a meta level, so probably at risk of leading nowhere in any reasonable amount of time :) However I hope that exploring it a bit can help us to understand better how other people in the Commons are thinking about this space.

[^1]: See this [example](https://github.com/InnerSourceCommons/InnerSourcePatterns/issues/542) of such a conversation.

## Conversation starter

We define InnerSource as:
> the use of open source best practices for software development within the confines of an organization.

So when producing material for the InnerSource Commons (e.g. a Pattern), I see two approaches on the extremes:
1. saying "just do what they do in open source" and possibly linking to a resource outside of the Commons
2. repeating everything that is already documented elsewhere

As in so many cases, the extremes are often not that helpful :)

Some assumptions that I make about the audience consuming the material produced by the Commons:
- the reader may not have much hands-on experience with open source (i.e. the details of how an open source project works on the inside are unknown)
- even if they known the open source mechanics a bit better, they may need help with translating those concepts into the company-internal context

Now what do we do with this? What should be our mindset when producing material in the Commons?

Questions that we could ask ourselves, when trying to explain a given InnerSource concept:
- which documentation about this concept exists in other domains?
- which of that material can we link to (e.g. in open source), preferably freely link to (i.e. the material is not behind a paywall)
- does the documentation about this concept from other domains need "translation"? i.e. are there key differences in how this is applied in InnerSource? is it worthwhile to point out these differences in writing?
- is the concept we are explaining strictly tied to InnerSource, or is it just "good engineering practices"? in other words what is the dividing line between what is InnerSource and what is not?
 
As you can tell, these are unfinished thoughts. Looking forward to the conversation.

**Comment by rrrutledge** [comment]
This point is a good one to bring up, @spier.  I've thought about this in the past and areas where InnerSource overlaps with other areas that have established literature and patterns.  Here is a diagram: 
![image](https://github.com/InnerSourceCommons/InnerSourcePatterns/assets/9609562/8cda731d-30ae-4b98-ae7d-bf747ade4062)
Here are my favorite resources for each area:
* [InnerSource Patterns](https://patterns.innersourcecommons.org/)
* [Open Source Guide](https://opensource.guide/)
* [Fearless Change Patterns](https://fearlesschangepatterns.com/)
* [Art of Community](https://www.jonobacon.com/books/artofcommunity/)

You bring up a 5th area, which is good engineering practices.  The overlapping diagram is going to [get complicated](https://www.bing.com/images/search?view=detailV2&ccid=sU33I9nF&id=8D4D9A32F4549C36E1FC671AD938D8548AB955A3&thid=OIP.sU33I9nFtNU2Clz9tCg2HwHaHa&mediaurl=https%3a%2f%2fi.stack.imgur.com%2ff1o3b.png&cdnurl=https%3a%2f%2fth.bing.com%2fth%2fid%2fR.b14df723d9c5b4d5360a5cfdb428361f%3frik%3do1W5ilTYONkaZw%26pid%3dImgRaw%26r%3d0&exph=480&expw=480&q=5+way+venn+diagram&simid=608011797356289655&FORM=IRPRST&ck=2E7839E1A46F718FF0D55EC2C60CB8DB&selectedIndex=0&ajaxhist=0&ajaxserp=0)!

I think that when producing some new content for InnerSource it behooves us to stop and think about any likely overlap with other areas that may already have established, great content.  After reviewing, then decide to link to it, copy it, translate it, reference it, or keep it as background and write something completely new.  Maybe there could be a decision tree process for deciding what to do based on what you find.
